src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -22,34 +23,54 @@ interface CSVOptions {
   escapeFormulaValues: boolean;
   formatFactory: FormatFactory;
   raw?: boolean;
-  columnsSorting?: string[];
+  sortedColumns?: string[];
+  columnSorting?: EuiDataGridColumnSortingConfig[];
 }
 
 export function datatableToCSV(
   { columns, rows }: Datatable,
-  { csvSeparator, quoteValues, formatFactory, raw, escapeFormulaValues, columnsSorting }: CSVOptions
+  {
+    csvSeparator,
+    quoteValues,
+    formatFactory,
+    raw,
+    escapeFormulaValues,
+    sortedColumns,
+    columnSorting,
+  }: CSVOptions
 ) {
   const escapeValues = createEscapeValue({
     separator: csvSeparator,
     quoteValues,
     escapeFormulaValues,
   });
 
-  const sortedIds = columnsSorting || columns.map((col) => col.id);
+  const sortedIds = sortedColumns
+    ? columns
+        .map((c) => {
+          // need to find original id for transposed column
+          const sortIndex = sortedColumns.findIndex((id) => c.id.endsWith(id));
+          return {
+            id: c.id,
+            sortIndex,
+            isTransposed: (sortedColumns[sortIndex] ?? '') !== c.id,
+          };
+        })
+        .filter(({ sortIndex }) => sortIndex >= 0)
+        // keep original zipped order between multiple transposed columns
+        .sort((a, b) => (a.isTransposed && b.isTransposed ? 0 : a.sortIndex - b.sortIndex))
```

Comment:
When there is a **Split metrics by** metric with 2+ **Metrics** applied, the datatable behavior implicitly groups like values together before respecting the column sorting order. This keeps the original order between all transposed columns.

<img width="1218" alt="image" src="https://github.com/user-attachments/assets/d32cbee3-68b2-4c67-8aea-4da395702ae3">

Reply:
This is now handled in `vis. getExportDatatables`